src/chains/ethereum/ethereum/src/forking/fork.ts

```diff
@@ -105,11 +104,24 @@ export class Fork {
   #setBlockDataFromChainAndOptions = async () => {
     const options = this.#options;
     if (options.blockNumber === Tag.LATEST) {
-      // if our block number option is "latest" override it with the original
-      // chain's current blockNumber
-      const block = await fetchBlock(this, Tag.LATEST);
+      const latestBlock = await fetchBlock(this, Tag.LATEST);
+      let blockNumber = parseInt(latestBlock.number, 16);
+      const currentTime = BigInt((Date.now() / 1000) | 0); // current time in seconds
```

Comment:
This logic is actually broken because:

1) a user's system clock can't be relied on.
2) block timestamps are permitted to be up to 900 seconds after the previous block (so basically 900 seconds in the future)

We could use a public NTP server to to sync the time, which isn't hard... but it now requires an Internet connection :-(

We should probably drop this in favor of a `"latest" - n` solution, where `n` is maybe `5 ` or so (configurable via options, of course).

Reply:
After talking it over with @MicaiahReid, we've decided that we'll switch to the `"latest" - n` idea, but only fork "known" chain ids (chain id's that ethereumjs-common knows about). For all other chain id's we will just use the `"latest"` number itself.